src/ansys/motorcad/core/methods/adaptive_geometry.py

```diff
@@ -0,0 +1,132 @@
+"""Methods for adaptive geometry."""
+from ansys.motorcad.core.geometry import Region
+
+
+class _RpcMethodsAdaptiveGeometry:
+    def __init__(self, mc_connection):
+        self.connection = mc_connection
+
+    def set_adaptive_parameter_value(self, name, value):
+        """Set adaptive parameter, if parameter does not exist then add it.
+
+        Parameters
+        ----------
+        name : string
+            name of parameter.
+        value : float
+            value of parameter.
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        method = "SetAdaptiveParameterValue"
+        params = [name, value]
+        return self.connection.send_and_receive(method, params)
+
+    def get_adaptive_parameter_value(self, name):
+        """Get adaptive parameter.
+
+        Parameters
+        ----------
+        name : string
+            name of parameter.
+
+        Returns
+        -------
+        float
+            value of parameter.
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        method = "GetAdaptiveParameterValue"
+        params = [name]
+        return self.connection.send_and_receive(method, params)
+
+    def get_region(self, name):
+        """Get Motor-CAD geometry region.
+
+        Parameters
+        ----------
+        name : string
+            name of region.
+
+        Returns
+        -------
+        ansys.motorcad.core.geometry.Region
+            Motor-CAD region object.
+
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        method = "GetRegion"
+        params = [name]
+        raw_region = self.connection.send_and_receive(method, params)
+
+        region = Region()
+        region._from_json(raw_region)
+
+        return region
+
+    def set_region(self, region):
+        """Set Motor-CAD geometry region.
+
+        Parameters
+        ----------
+        region : ansys.motorcad.core.geometry.Region
+            Motor-CAD region object.
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        raw_region = region._to_json()
+
+        method = "SetRegion"
+        params = [raw_region]
+        return self.connection.send_and_receive(method, params)
+
+    def check_closed_region(self, region):
+        """Check region is closed using region detection.
+
+        Parameters
+        ----------
+        region : ansys.motorcad.core.geometry.Region
+            Motor-CAD region object.
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        pass
+
+    def check_collisions(self, region, regions_to_check):
```

Comment:
Is it worth testing this function?

Reply:
Unable to test at the moment, waiting for review.